lib/simulator-xcode-6.js

```diff
@@ -5,66 +5,119 @@ import { fs, util } from 'appium-support';
 import B from 'bluebird';
 import _ from 'lodash';
 import { utils as instrumentsUtil } from 'appium-instruments';
-import { killAllSimulators } from './util.js';
-import bplistParser from 'bplist-parser';
+import { killAllSimulators } from './utils.js';
+import _glob from 'glob';
+import { read } from './settings';
 
-let parseFile = B.promisify(bplistParser.parseFile);
 
-function getRootDir() {
-  let home = process.env.HOME;
+const glob = B.promisify(_glob);
 
-  return path.resolve(home, 'Library', 'Developer', 'CoreSimulator', 'Devices');
-}
 
 class SimulatorXcode6 {
-
   constructor (udid, xcodeVersion) {
+    this.udid = String(udid);
     this.xcodeVersion = xcodeVersion;
-    this.udid = udid;
+
+    // platformVersion cannot be found initially, since getting it has side effects for
+    // our logic for figuring out if a sim has been run
+    // it will be set when it is needed
+    this.platformVersion = null;
 
     //TODO assert that udid is a valid sim
-    this.keychainPath = path.resolve(this.getDir(), "Library", "Keychains");
+    this.keychainPath = path.resolve(this.getDir(), 'Library', 'Keychains');
+
+    this.appDataBundlePaths = null;
   }
 
   //TODO default constructor with no args should create a sim with some defaults
 
+  async getPlatformVersion () {
```

Comment:
what about making this `async get platformVersion()`, then storing the actual value in an inaccessible cached version?